src/core/issues-notes.js

```diff
@@ -146,50 +170,89 @@ function handleIssues($ins, ghIssues, conf) {
   }
 }
 
-async function fetchIssuesFromGithub({ githubAPI }) {
-  const issues = [];
-  const issueNumbers = [...document.querySelectorAll(".issue[data-number]")]
-    .map(elem => Number.parseInt(elem.dataset.number, 10))
-    .filter(number => number);
-  for (const issueNumber of issueNumbers) {
-    let issue; // depends on how fetching goes...
+//derives the text-color-class based on the illumination score of the labelColor
+function deriveTextColorClass(hexColor) {
+  const rgb = parseInt(hexColor, 16);
+  if (isNaN(rgb)) {
+    throw new TypeError(`Invalid hex color: ${hexColor}`);
+  }
+  const red = (rgb >> 16) & 0xff;
+  const green = (rgb >> 8) & 0xff;
+  const blue = (rgb >> 0) & 0xff;
+  const illumination = 0.2126 * red + 0.7152 * green + 0.0722 * blue;
+  return illumination > 140 ? "light" : "dark";
+}
+
+function createLabel(label) {
+  const { color, href, name } = label;
+  let textColorClass = "dark";
+  try {
```

Comment:
I think this is an incorrect use of try-catch, we can just `const textColorClass = isNaN(color) ? "dark" : deriveTextColorClass(color);`.

Reply:
ok, yeah... so let's unwrap this and ensure `deriveTextColorClass` only takes numbers.